Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainActivity.java

```diff
@@ -259,4 +246,33 @@ private PlatformGamesView getPlatformGamesView(Platform platform)
 
     return (PlatformGamesView) getSupportFragmentManager().findFragmentByTag(fragmentTag);
   }
+
+  // Don't call this before DirectoryInitialization completes.
+  private void setPlatformTabsAndStartGameFileCacheService()
+  {
+    PlatformPagerAdapter platformPagerAdapter = new PlatformPagerAdapter(
+            getSupportFragmentManager(), this);
+    mViewPager.setAdapter(platformPagerAdapter);
+    mViewPager.setOffscreenPageLimit(platformPagerAdapter.getCount());
+    mTabLayout.setupWithViewPager(mViewPager);
+    mTabLayout.addOnTabSelectedListener(new TabLayout.ViewPagerOnTabSelectedListener(mViewPager)
+    {
+      @Override
+      public void onTabSelected(@NonNull TabLayout.Tab tab)
+      {
+        super.onTabSelected(tab);
+        NativeLibrary
+                .SetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_ANDROID,
+                        SettingsFile.KEY_LAST_PLATFORM_TAB, Integer.toString(tab.getPosition()));
+      }
+    });
+
+    String platformTab = NativeLibrary
+            .GetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_ANDROID,
+                    SettingsFile.KEY_LAST_PLATFORM_TAB, "0");
+
+    mViewPager.setCurrentItem(Integer.parseInt(platformTab));
```

Comment:
You should probably have a fallback here (e.g. selecting the first tab) if `parseInt` fails. You never know what kinds of weird things users might end up with in their INIs.

Reply:
Would a try-catch be best here? I know one should avoid using them if possible but it seems like that is a good choice here.

Reply:
That sounds fine to me. But please catch `NumberFormatException` instead of `Exception`, so that it's clearer what the purpose of the try-catch is.